### PR TITLE
chore(deps): upgrade to Go 1.22.2

### PR DESCRIPTION
**Pull request by rootulp**
Closes https://github.com/celestiaorg/celestia-app/issues/3262

### CODE REVIEW

File: test/testground/go.mod [hunk 1]
```diff
@@ -191,14 +191,14 @@ require (
 	go.uber.org/atomic v1.11.0 // indirect
 	go.uber.org/multierr v1.11.0 // indirect
 	go.uber.org/zap v1.24.0 // indirect
-	golang.org/x/crypto v0.18.0 // indirect
+	golang.org/x/crypto v0.19.0 // indirect
 	golang.org/x/exp v0.0.0-20231206192017-f3f8817b8deb // indirect
 	golang.org/x/mod v0.14.0 // indirect
-	golang.org/x/net v0.20.0 // indirect
+	golang.org/x/net v0.21.0 // indirect
```
Comment (cmwaters) [inline test/testground/go.mod:197]: I thought we wanted this to be `golang.org/x/net v0.23.0` to avoid the go vuln warning

**Review by cmwaters — DISMISSED** [review]
lgtm

File: test/testground/go.mod (hunk 1 above)
Comment (rootulp) [reply test/testground/go.mod:197]: we doooo, good catch. I fixed in the main go.mod but pretty sure I forgot to run `go work sync` after that

Comment (rootulp) [reply test/testground/go.mod]: https://github.com/celestiaorg/celestia-app/pull/3263/commits/9f19eceb1a5f404854f630780ab128efff7ba41c

**Review by staheri14 — APPROVED** [review]
Thanks!

**Review by cmwaters — APPROVED** [review]

**Review by ninabarbakadze — APPROVED** [review]

**Pull request merged**